src/core/service.c

```diff
@@ -690,6 +699,9 @@ static int service_verify(Service *s) {
         if (s->type == SERVICE_DBUS && !s->bus_name)
                 return log_unit_error_errno(UNIT(s), SYNTHETIC_ERRNO(ENOEXEC), "Service is of type D-Bus but no D-Bus service name has been specified. Refusing.");
 
+        if (!UNIT(s)->transient && get_n_extra_fds(s) > 0)
+                return log_unit_error_errno(UNIT(s), SYNTHETIC_ERRNO(ENOEXEC), "Service has ExtraFileDescriptor= set, which isn't allowed for non-transient services. Refusing.");
```

Comment:
Should use plural here. Though this might be a bit too paranoid since we don't have a unit file parser for the setting so it can only ever be set via dbus in the first place no?

Reply:
Yes that's true, I just added this for extra insurance in case code paths get refactored and somehow someone could set this from a unit file. Happy to remove it if its overkill

Reply:
Ended up removing it!